lib/javascripts/base_app.js

```diff
@@ -68,6 +68,10 @@ function BaseApp(zafClient, data) {
   let evt = { firstLoad: true };
   resolveHandler(this, 'app.created')();
   resolveHandler(this, 'app.activated')(evt, evt);
+  $(window).unload(() => {
+    resolveHandler(this, 'app.willDestroy')();
+    delete this.events['app.willDestroy']; // make sure this event only fires once
```

Comment:
> Or we could simply skip `willDestroy` when binding events.

I think the true fix is in the framework, this event should not be fired into v2 apps at all.